homeassistant/components/switchgrid/config_flow.py

```diff
@@ -0,0 +1,35 @@
+"""Config flow for Switchgrid integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+STEP_USER_DATA_SCHEMA = vol.Schema({})
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Switchgrid."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is not None:
+            return self.async_create_entry(title="Switchgrid", data=user_input)
+
+        return self.async_show_form(step_id="user")
```

Comment:
We should test if we can fetch data and abort properly

Reply:
@joostlek You mean check whether the coordinator is able to successfully fetch its data, and if not, then abort the process of installing the integration?

I'm not sure how to access that information from there. Should I instantiate the coordinator here?

Sorry, I'm new to python and the internals of HomeAssistant, too.